packages/cli/src/lib/commands/collect/utils/persist/open-report.unit.test.ts

```diff
@@ -0,0 +1,108 @@
+import * as openReport from 'open';
+import { handleOpenFlowReports, openFlowReports } from './open-report';
+import { logVerbose } from '../../../../core/loggin';
+import { CollectCommandOptions } from '../../options';
+
+describe('handleOpenFlowReport', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  })
+
+  it('should return the openFlowReport function if openReport, interactive and not dryRun', async () => {
+    const openReportsProcess = handleOpenFlowReports({
+      openReport: true,
+      interactive: true,
+      dryRun: false,
+    } as CollectCommandOptions);
+    expect(typeof openReportsProcess).toEqual("function");
+  });
+
+  it('should return undefined if openReport is false', () => {
+    const openReportsProcess = handleOpenFlowReports({
+      openReport: false,
+      interactive: true,
+      dryRun: false,
+    } as CollectCommandOptions);
+    expect(openReportsProcess).toEqual(undefined);
+  });
+
+  it('should return undefined if dryRun is true', () => {
+    const openReportsProcess = handleOpenFlowReports({
+      openReport: true,
+      interactive: true,
+      dryRun: true,
+    } as CollectCommandOptions);
+    expect(openReportsProcess).toEqual(undefined);
+  });
+
+  it('should return undefined if interactive is false', () => {
+    const openReportsProcess = handleOpenFlowReports({
+      openReport: true,
+      interactive: false,
+      dryRun: false,
+    } as CollectCommandOptions);
+    expect(openReportsProcess).toEqual(undefined);
+  });
+
+
+});
+
+jest.mock('open');
+jest.mock('../../../../core/loggin');
+
+describe('openReports', () => {
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should not open the report if no file name is passed', async () => {
+    await openFlowReports([]);
+    expect(openReport).not.toHaveBeenCalled();
+  });
+
+  it('should not logVerbose if no file name is passed', async () => {
+    await openFlowReports([]);
+    expect(logVerbose).not.toHaveBeenCalled();
+  });
+
+  it('should open the report if filenames includes an html report', async () => {
+    await openFlowReports(['example.html']);
+    expect(openReport).toHaveBeenCalled();
+  });
+
+  it('should logVerbose if filenames includes an html report', async () => {
+    await openFlowReports(['example.html']);
+    expect(logVerbose).toHaveBeenCalledWith(expect.stringContaining('HTML'));
+  });
```

Comment:
I wonder if it wouldn't be better to add the `logVerbose` checks in the `should open report` test case (or leave it out). Or is it an important part of the logic that deserves its own use case?

I am by no means familiar enough with the critical flows of this application to decide this so feel free to ignore if this is not applicable to you.

**More context**
There can be a trap for a developer to include all states a component or a function can get into. The "well since this statement/branching logic is here so I should cover it with tests". Then the test cases are also named after the implementation states as opposed to use cases.

This is why TDD can be very valuable (aka designing tests first, writing implementation to pass the tests second) or someone else (who did not implement the feature) helping design the tests - one is not tempted to just cover the current implementation but focuses on the important use cases the implementation should accommodate.